File: VB/AuthenticationOwin.Module/Security/CustomAuthenticationStandardProvider.vb
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Security

Namespace AuthenticationOwin.Module.Security
    Public Class CustomAuthenticationStandardProvider
        Inherits AuthenticationStandardProvider
        Public Sub New(ByVal userType As Type)
            MyBase.New(userType)
        End Sub
        Public Overrides Function Authenticate(ByVal objectSpace As IObjectSpace) As Object
            Dim user As IAuthenticationOAuthUser = TryCast(MyBase.Authenticate(objectSpace), IAuthenticationOAuthUser)
            If user IsNot Nothing AndAlso (Not user.EnableStandardAuthentication) Then
                Throw New InvalidOperationException("Password authentication is not allowed for this user.")
            End If
            Return user
        End Function
    End Class
End Namespace

File: CS/AuthenticationOwin.Module/Security/CustomAuthenticationStandardProvider.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Security;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AuthenticationOwin.Module.Security {
    public class CustomAuthenticationStandardProvider : AuthenticationStandardProvider {
        public CustomAuthenticationStandardProvider(Type userType) : base(userType) {
        }
        public override object Authenticate(IObjectSpace objectSpace) {
            IAuthenticationOAuthUser user = base.Authenticate(objectSpace) as IAuthenticationOAuthUser;
            if(user != null && !user.EnableStandardAuthentication) {
                throw new InvalidOperationException("Password authentication is not allowed for this user.");
            }
            return user;
        }
    }
}
